feature/coc blog preload

iamscottcab opened the pull request:
Added code of conduct link and initial blog content re: website release and future plans.

Commit:
feat(coc-blog-preload): added coc page, made blog hero image optional, added author image

Commit:
feat(coc-blog-preload): added initial blog post

dan-harris commented on src/blog/posts/forward-to-2020.md:
I think we need to add a 'heapsgood' author. That way itll spit out the actual author + this being an 'official' heapsgood blog post. Or we could do it some other way (a boolean frontmatter or something)

iamscottcab replied:
I have added one now, if this conflicts with your style tweaks (if you added a new profile) we might be able to blow away the one I added but the new blog post includes hgd as the first author.

dan-harris commented on src/blog/posts/forward-to-2020.md:
🙌

dan-harris commented on src/blog/posts/forward-to-2020.md:
Not sure if we need to re-phrase this. Will discuss 🙂

iamscottcab replied:
Haven't heard back about this on Slack but we can tweak it once it's in dev.

dan-harris reviewed (commented):
I think the splash image might need a bit of manual optimising (if possible). Ideally we should set this up to be automatic.... 🤷‍♀️

Commit:
feat(coc-blog-preload): added a heroimage tag to blog front matter

this allows us to specify the file type of the image and use images with a different name (maybe
placeholder images etc) instead of relying on the slug name and png file format.

iamscottcab commented:
> I think the splash image might need a bit of manual optimising (if possible). Ideally we should set this up to be automatic.... 🤷‍♀️

I've separated the blog image from the slug. There is a new property in front matter for the blog called `heroimage` which represents the name and filetype of the image. Was able to conver the 1.28mb png to a BrowserJPG that was 94% reduction in size.

dan-harris commented on src/_includes/_nav.html:
```diff
@@ -15,10 +15,10 @@
-          <a class="navbar__main-nav__link{{"active" if (base_route == "presentations")}}"
-            href="/presentations">slides</a>
-        </li> #}
+        <li class="navbar__main-nav__link__outer">
```
Not hugely important at this stage, but I think the code of conduct can sit outside the main nav (especially as we add other sections)

iamscottcab replied:
I think in the interim it definitely needs to be here especially because we don't have a footer or anywhere else to put it and I don't think it should be buried in the blog somewhere. Happy to re-evaluate it once we get more top nav though.

dan-harris commented on src/_layouts/blog.njk:
```diff
@@ -6,11 +6,14 @@ layout: page-default
 <article class="blog-post content__inner--full-width">
     <!-- header title + image .etc -->
     <header class="blog-post__header">
-        <div class="blog-post__header__image">
```
I think, however we do it, the hero image should always be required.

iamscottcab replied:
If you want this to always be required then we need to spin up another layout for the blog or other static pages where a hero image is not required. In this instance I think having one of the CoC is a bit weird which is why I made it variable but we can spin out other layout as an enhancement if it is more explicit.

dan-harris commented on src/blog/posts/forward-to-2020.md:
```diff
@@ -0,0 +1,49 @@
+---
+title: Looking forward to 2020
+heroimage: forward-to-2020.jpg
```
after thinking about this a little, I think it might be best to leave this implicit (shock horror coming from me i know 😀). that was it forces our images to have a certain name structure. Can just change our default file extension to jpg? I'm not going to get super hung up on it, so if you disagree, just go for it 🙂

Also, we need to update the `og:` tags in the header if we do change the image frontmatter.

iamscottcab replied:
I have returned this back to implicit to save retesting social links.

dan-harris reviewed (commented):
A few more comments, see what you think. if they don't matter, then just resolve them and merge 🤷‍♀️ (main one is to update the social header tags to use the new image)

Commit:
feat(coc-blog-preload): fixed authors, slugs links in CoC & blog post

Commit:
feat(coc-blog-preload): added author data and cleaned up sample blog page
